### ISSUE TITLE
cat snapshots API requires repository name

### ISSUE THREAD

**Issue by peterpramb** [elastic/elasticsearch#50181]
**Elasticsearch version** (`bin/elasticsearch --version`):
`Version: 7.5.0, Build: default/tar/e9ccaed468e2fac2275a3761849cbee64b39519f/2019-11-26T01:06:52.518245Z, JVM: 11.0.5`

**Plugins installed**:
`repository-s3`

**JVM version** (`java -version`):
`OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.5+10, mixed mode)`

**OS version** (`uname -a` if on a Unix-like system):
`Linux appaidf1 3.10.0-957.21.3.el7.x86_64 #1 SMP Fri Jun 14 02:54:29 EDT 2019 x86_64 x86_64 x86_64 GNU/Linux`

**Description of the problem including expected versus actual behavior**:
According to the [documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/cat-snapshots.html) the `cat snapshots API` should return the content of all repositories, if no specific repository or `_all` is given. Unfortunately neither one works and just produce errors.

Not sure if this is a bug or a documentation issue.

**Steps to reproduce**:
```
GET /_cat/repositories
plgrels7-at-fat s3
```
```
GET /_cat/snapshots
{
    "error": {
        "reason": "Validation Failed: 1: repository is missing;",
        "root_cause": [
            {
                "reason": "Validation Failed: 1: repository is missing;",
                "type": "action_request_validation_exception"
            }
        ],
        "type": "action_request_validation_exception"
    },
    "status": 400
}
```
```
GET /_cat/snapshots/_all
{
    "error": {
        "reason": "[_all] missing",
        "root_cause": [
            {
                "reason": "[_all] missing",
                "type": "repository_missing_exception"
            }
        ],
        "type": "repository_missing_exception"
    },
    "status": 404
}
```
```
GET /_cat/snapshots/plgrels7-at-fat
weekly-2019.11.28-1mpcnufqsgq1fnp2lrp3da SUCCESS 1574977884 21:51:24 1574978190 21:56:30  5.1m 40 40 0 40
weekly-2019.11.29-vohughnnrqmde5dmr69ggw SUCCESS 1575057600 20:00:00 1575057629 20:00:29 29.4s 41 41 0 41
weekly-2019.12.06-ozp2fuuss46awjkjzkfiyg SUCCESS 1575662400 20:00:00 1575662446 20:00:46 46.4s 47 47 0 47
```

**Comment by original-brownbear** [comment]
This is a bug in the docs that seems to have been introduced by a8e027551ad66eb4ccee6e83c9b075b3e34c2738 . The `_cat` across all repos will only work in `master/v8` but not in any current release.

**Comment by jrodewig** [comment]
Closed by #50274. Thanks @peterpramb @original-brownbear
